crates/contract/tests/sandbox/common.rs

```diff
@@ -761,35 +810,38 @@ pub async fn submit_ckd_response(
     Ok(())
 }
 
-pub async fn sign_and_validate(
-    account: &Account,
-    request: &SignRequestArgs,
-    respond: Option<&SignResponseArgs>,
+pub async fn request_is_in_queue(
     contract: &Contract,
-    attested_account: &Account,
-) -> anyhow::Result<()> {
-    let status = submit_sign_request(account, request, contract).await?;
-
-    tokio::time::sleep(std::time::Duration::from_secs(3)).await;
+    request: &SignatureRequest,
+) -> Option<YieldIndex> {
+    contract
+        .view("get_pending_request")
+        .args_json(serde_json::json!({"request": request}))
+        .await
+        .unwrap()
+        .json()
+        .unwrap()
+}
 
-    if let Some(response) = respond {
-        submit_signature_response(response, contract, attested_account).await?;
-    }
+pub async fn wait_for_request(
+    contract: &Contract,
+    request: &SignatureRequest,
+    max_timeout: Option<Duration>,
+) -> anyhow::Result<()> {
+    let timeout = max_timeout.unwrap_or(Duration::from_secs(3));
+    let start = std::time::Instant::now();
 
-    let execution = status.await?;
-    dbg!(&execution);
-    let execution = execution.into_result()?;
+    loop {
+        if request_is_in_queue(contract, request).await.is_some() {
+            return Ok(());
+        }
 
-    // Finally wait the result:
-    let returned_resp: SignatureResponse = execution.json()?;
+        if start.elapsed() >= timeout {
+            anyhow::bail!("timed out waiting for request to appear in queue");
+        }
 
-    if let Some(response) = respond {
-        assert_eq!(
-            &returned_resp, &response.response,
-            "Returned signature request does not match"
-        );
+        tokio::time::sleep(Duration::from_millis(200)).await;
```

Comment:
isn't this too big? Or this is supposed to take an amount of time in the order of seconds? My worry is that if a test uses a chain of many requests then waiting this long would make it much slower. The operation inside the loop does not seem to inefficient, so a lower sleep, maybe 50ms or less seems better IMO